assets/src/edit-story/utils/useAdminMenuCollapsed.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useState } from '@wordpress/element';
+
+const useAdminMenuCollapsed = () => {
+	const [ isCollapsed, setIsCollapsed ] = useState( false );
+
+	useEffect( () => {
+		if ( document.body.classList.contains( 'folded' ) ) {
+			setIsCollapsed( true );
+		}
+
+		const { MutationObserver } = window;
+		const observer = new MutationObserver( ( mutations ) => {
+			mutations.forEach( ( mutation ) => {
+				if (
+					( mutation.oldValue.match( /folded/ ) && ! mutation.target.classList.contains( 'folded' ) ) ||
+					( mutation.oldValue.match( /folded/ ) && mutation.target.classList.contains( 'folded' ) )
+				) {
+					setIsCollapsed( ! isCollapsed );
+				}
+			} );
+		} );
+
+		if ( document.body ) {
+			observer.observe( document.body, {
+				attributes: true,
+				attributeOldValue: true,
+				attributeFilter: [ 'class' ],
+			} );
+		}
+
+		return () => {
+			observer.disconnect();
+		};
+
+	// Disable reason: We want this hook to fire only once (like componentDidMount).
```

Comment:
Which dependencies are missing here? If it's only `isCollapsed`, then you can simply change the setter to toggle using the functional variant:

```
setIsCollapsed( ( isCollapsed ) => ! isCollapsed );
```